### PR TITLE
DAOS-3419: remove health stats from storage scan output

### PR DESCRIPTION
**Pull request by tanabarr**
NVMe health statistics can be returned by using the "dmg storage
query nvme-health" subcommand which will return controller and
namespace details in addition to health statistics.

In order to make the "{dmg,daos_server} storage scan" output more
digestible, remove health statistics from the NVMe output.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

File: src/control/common/storage/pb_types.go [hunk 1]
```diff
@@ -30,94 +30,116 @@ import (
...
+type NvmeHealthstats []*ctlpb.NvmeController_Health
+
+func (stats NvmeHealthstats) String() string {
+	var buf bytes.Buffer
+
+	for _, stat := range stats {
+		fmt.Fprintf(&buf, "\tHealth Stats:\n\t\tTemperature:%dK(%dC)\n", stat.Temp, stat.Temp-273)
+
+		if stat.Tempwarn > 0 {
+			fmt.Fprintf(&buf, "\t\t\tWarning Time:%d\n", uint64(stat.Tempwarn))
```
[the remark this answers is not in the document]
Comment (mjmac) [reply src/control/common/storage/pb_types.go]: If we can figure out what unit the value represents we can wrap it with a time.Duration which provides a nice stringer implementation.

e.g. https://play.golang.org/p/rYZJJiv8Jdk
Temperature Warning Duration: 1m40s

[the remark this answers is not in the document]
Comment (sydneyvanda) [reply src/control/common/storage/pb_types.go]: This is warning time is in minutes

File: src/control/cmd/dmg/storage.go [hunk 2]
```diff
@@ -84,16 +83,14 @@ type storageScanCmd struct {
...
-	log.Infof("NVMe SSD controllers and constituent namespaces:\n%s", cNvmeScan)
-	log.Info(cScmScan.String())
-}
-
-// Execute is run when storageScanCmd activates
+// Execute is run when storageScanCmd activates.
+// Runs NVMe and SCM storage scan on all connected servers.
 func (s *storageScanCmd) Execute(args []string) error {
-	storageScan(s.log, s.conns)
+	req := client.StorageScanReq{NvmeHealth: false}
```
Comment (sydneyvanda) [inline src/control/cmd/dmg/storage.go]: Is the plan to make the health stats optional on scan at some point? I was under the impression the health stats were only going to be queried by the nvme-health command.

File: src/control/common/storage/pb_types.go [hunk 3]
```diff
@@ -30,90 +30,104 @@ import (
...
-			"\tPCI Addr:%s Serial:%s Model:%s Fwrev:%s Socket:%d\n",
-			ctrlr.Pciaddr, ctrlr.Serial, ctrlr.Model, ctrlr.Fwrev,
-			ctrlr.Socketid)
+	fmt.Fprintf(buf, "\tHealth Stats:\n\t\tTemperature:%dK(%dC)\n", stat.Temp, stat.Temp-273)
 
-		for _, ns := range ctrlr.Namespaces {
-			fmt.Fprintf(&buf, "\t\tNamespace: %+v\n", ns)
-		}
+	if stat.Tempwarn > 0 {
+		fmt.Fprintf(buf, "\t\t\tWarning Time:%d\n", uint64(stat.Tempwarn))
```
Comment (sydneyvanda) [inline src/control/common/storage/pb_types.go]: We should clarify with minutes here, as @kjacque suggested

File: src/control/common/storage/pb_types.go [hunk 4]
```diff
@@ -30,90 +30,104 @@ import (
...
+	}
+	if stat.Tempcrit > 0 {
+		fmt.Fprintf(buf, "\t\t\tCritical Time:%d\n", uint64(stat.Tempcrit))
```
Comment (sydneyvanda) [inline src/control/common/storage/pb_types.go]: Same as above

File: src/control/cmd/dmg/storage.go (hunk 2 above)
Comment (mjmac) [reply src/control/cmd/dmg/storage.go]: Yes, that is the goal. We're still working on the details of how these tools will display output, but my preference would be for the health stats to only be displayed on scan if values are out of their normal ranges.

Comment (tanabarr) [reply src/control/cmd/dmg/storage.go]: the flag in the storage scan request signifies whether or not the output contains the health information, the nvme-health subcommand calls StorageScan with the flag in the request set to true

**Pull request merged**